qiime2/metadata/tests/test_metadata.py

```diff
@@ -1625,20 +1556,22 @@ def test_no_artifacts(self):
         self.assertEqual(filtered.artifacts, ())
 
     def test_with_artifacts(self):
-        artifact1 = Artifact.import_data('Mapping', {'a': '1', 'b': '2'})
-        artifact2 = Artifact.import_data('Mapping', {'d': '4'})
+        artifact1 = Artifact.import_data('Mapping', {'col1': [1, 2, 3]})
+        artifact2 = Artifact.import_data(
+            'Mapping', {'col2': ['foo', 'bar', 'baz']})
 
         md = Metadata(pd.DataFrame(
             {'col1': [1, 2, 3], 'col2': ['foo', 'bar', 'baz']},
             index=pd.Index(['a', 'b', 'c'], name='id')))
-        md._add_artifacts([artifact1, artifact2])
+        md._column_sources = {'col1': artifact1,
+                              'col2': artifact2}
 
         obs = md.filter_columns(column_type='categorical')
 
         exp = Metadata(pd.DataFrame(
             {'col2': ['foo', 'bar', 'baz']},
             index=pd.Index(['a', 'b', 'c'], name='id')))
-        exp._add_artifacts([artifact1, artifact2])
+        exp._add_artifacts(artifact1, artifact2)
```

Comment:
This goes back to the "should the filters remove filtered columns from `_column_sources`" deal. If filtered columns are removed from `column_sources` then `artifact1` is no longer in the expected metadata's `artifacts` property.

Reply:
Please see my comment, above.